cluster/network.go

```diff
@@ -0,0 +1,23 @@
+package cluster
+
+import (
+	"fmt"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/rancher/rke/k8s"
+	"github.com/rancher/rke/pki"
+)
+
+func (c *Cluster) DeployNetworkPlugin() error {
+	logrus.Debugf("[Network] Setting network plugin: %s", c.NetworkPlugin)
+	deployerHost := c.ControlPlaneHosts[0]
```

Comment:
We should add a guard for validation the size of the control plane hosts. What if there are none? Up to you when you add it, may be in the place where the cluster.yml file is being parsed?